src/module.cpp

```diff
@@ -3029,10 +3092,23 @@ int Module::CompileAndOutput(const char *srcFile, Arch arch, const char *cpu, st
         lEmitDispatchModule(dispatchModule, exportedFunctions);
 
         if (outFileName != NULL) {
-            if ((outputType == Bitcode) || (outputType == BitcodeText))
+            switch (outputType) {
+            case CPPStub:
+                break;
+
+            case Bitcode:
+            case BitcodeText:
                 writeBitcode(dispatchModule, outFileName, outputType);
```

Comment:
Let's do proper error reporting here, i.e.
```c++
if (!writeBitcode(dispatchModule, outFileName, outputType))
    return 1;
```

Same for `writeObjectFileOrAssembly` below